ifrs17/Import/Importers.ipynb

```diff
@@ -951,19 +993,22 @@
                 "\n            var estimateType = isEstimateType ? amountTypeFromFile : EstimateTypes.BE;",
                 "\n            ",
                 "\n            var values = datarow.Table.Columns.Where(c => c.ColumnName.StartsWith(nameof(RawVariable.Values))).OrderBy(c => c.ColumnName.Length).ThenBy(c => c.ColumnName)",
-                "\n                                .Select(x => datarow.Field<string>(x.ColumnName).CheckStringForExponentialAndConvertToDouble()).Prune();",
+                "\n                                .Select(x => datarow.Field<string>(x.ColumnName).CheckStringForExponentialAndConvertToDouble()).ToArray();",
                 "\n            ",
                 "\n            // Filter out empty raw variables for AocType != CL",
-                "\n            if(values.Length == 0 && aocType != AocTypes.CL) return null;  //TODO: extend this check for all mandatory step and not just for CL",
+                "\n            //TODO: extend this check for all mandatory step and not just for CL",
+                "\n            if(args.Scenario == null) {",
+                "\n                values = values.Prune();",
+                "\n                if(values.Length == 0 && aocType != AocTypes.CL) return null;",
+                "\n            }",
                 "\n                ",
                 "\n            var item = new RawVariable {",
                 "\n                DataNode = dataNode,",
                 "\n                AocType = aocType,",
                 "\n                Novelty = novelty,",
                 "\n                AmountType = amountType,",
                 "\n                EstimateType = estimateType,",
-                "\n                AccidentYear = dataset.Tables[ImportFormats.Cashflow].Columns.Where(x => x.ColumnName == nameof(RawVariable.AccidentYear)).Count()>0?",
-                "\n                            (Int32.TryParse((datarow.Field<string>(nameof(RawVariable.AccidentYear))), out var accidentYear)? accidentYear : (int?)null) : (int?)null, ",
+                "\n                AccidentYear = accidentYearColumn? (Int32.TryParse((datarow.Field<string>(nameof(RawVariable.AccidentYear))), out var accidentYear)? accidentYear : (int?)null) : (int?)null, ",
```

Comment:
Can we improve this line? This double (int?) is not nice.

Reply:
Proposal: 
change     
var accidentYearColumn = dataSet.Tables[ImportFormats.Cashflow].Columns.Where(x => x.ColumnName == nameof(RawVariable.AccidentYear)).Count() > 0;

=>
var missingAccidentYearColumn = dataSet.Tables[ImportFormats.Cashflow].Columns.Where(x => x.ColumnName == nameof(RawVariable.AccidentYear)).Count() == 0; 

and now: 
AccidentYear = missingAccidentYearColumn
                                ? (int?)null 
                                : Int32.TryParse((datarow.Field<string>(nameof(RawVariable.AccidentYear))), out var accidentYear)
                                    ? accidentYear 
                                    : (int?)null,

Reply:
What about this for the last part:
AccidentYear = (!missingAccidentYearColumn && Int32.TryParse((datarow.Field(nameof(RawVariable.AccidentYear))), out var accidentYear))
? accidentYear
: default

==> default should work, I think, because AccidentYear is defined to be int?. Otherwise, use (int?)null

Reply:
2 ternary operators are probably not needed.
can you say the following?
hasAccidentYearColumn && Int32.TryParse((datarow.Field(nameof(RawVariable.AccidentYear))), out var accidentYear) 
?  accidentYear
: something;

Reply:
implemented